src/main/java/com/mixpanel/android/mpmetrics/MixpanelNotificationRouteActivity.java

```diff
@@ -29,14 +29,31 @@ protected void onCreate(Bundle savedInstanceState) {
 
         trackAction(routeIntent);
 
-        Intent notificationIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(routeIntent.getExtras().getCharSequence("uri").toString()));
+        final Intent notificationIntent = handleRouteIntent(routeIntent);
+
+
         if (!extras.getBoolean("sticky")) {
             cancelNotification(extras);
         }
         startActivity(notificationIntent);
     }
 
-    private void cancelNotification(Bundle extras) {
+    protected Intent handleRouteIntent(Intent routeIntent) {
+        String actionType = routeIntent.getExtras().getCharSequence("actionType").toString();
+
+        if (actionType.equals("browser") || actionType.equals("deeplink")) {
```

Comment:
let's use an enum or at least constants for these action type values and use a switch/case instead of if/thens